include/zephyr/kernel.h

```diff
@@ -2927,12 +2929,12 @@ struct k_mutex {
  * @cond INTERNAL_HIDDEN
  */
 #define Z_MUTEX_INITIALIZER(obj) \
-	{ \
+	IF_ENABLED(CONFIG_MULTITHREADING, ({ \
```

Comment:
```suggestion
	{IF_ENABLED(CONFIG_MULTITHREADING, ( \
```

Reply:
might want to use `COND_CODE_1()` and set the "else" clause to `(0)` so that the initializer becomes `{0}`, which is possibly more MISRA-correct or something like that.

Reply:
Done!